boltstub/parsing.py

```diff
@@ -1138,8 +1141,17 @@ def can_consume_after_reset(self, channel) -> bool:
         return block.can_consume_after_reset(channel)
         pass
 
-    def has_deterministic_end(self):
-        return all(b.has_deterministic_end() for b in self.blocks)
+    def has_deterministic_end(self, channel=None) -> bool:
+        if channel is None:
+            if len(self.blocks) <= len(self.conditions):
+                # no else block => cannot guarantee deterministic end at static
+                # check time
+                return False
+            return all(b.has_deterministic_end(channel) for b in self.blocks)
```

Comment:
at this point (1150) we know `channel` is `None`, right? I think it might make it infinitesimally clearer to omit it, what do you think?